code/modules/mob/living/simple_animal/bot/cleanbot.dm

```diff
@@ -75,6 +154,9 @@
 				to_chat(user, "<span class='warning'>Please close the access panel before locking it.</span>")
 			else
 				to_chat(user, "<span class='notice'>\The [src] doesn't seem to respect your authority.</span>")
+	else if(istype(W, /obj/item/kitchen/knife) && user.a_intent != INTENT_HARM)
```

Comment:
\>cant strap a chainsaw to it

Reply:
The poor cleanbot would tip over! Those things are far too large!

..that being said Senator Cleanbot may come back later with a katana in future updates

Reply:
(you can just check for sharp objects instead of a particular typepath)

Reply:
Yeah but I only sprited a stubby little knife and it would be weird if you could tape a fire axe or circular saw to it. The original meme is just a roomba with a knife, so I'd prefer to just leave it there

Reply:
You could just grab the weapon's inhand sprite.

Reply:
>the original meme is just a roomba with a knife

You utter fool
https://www.youtube.com/watch?v=xrD8g1z8yg0